client/my-sites/ads/form-earnings.jsx

```diff
@@ -81,198 +71,229 @@ module.exports = React.createClass( {
 			showEarningsNotice: false,
 			showWordadsInfo: false,
 			showSponsoredInfo: false,
-			showAdjustmentInfo: false
+			showAdjustmentInfo: false,
 		} );
-	},
+	}
 
-	_fetchIfEmpty: function( site ) {
+	fetchIfEmpty( site ) {
 		site = site || this.props.site;
 		if ( ! site || ! site.ID ) {
 			return;
 		}
 
 		if ( EarningsStore.getById( site.ID ).earnings ) {
-			debug( 'initial fetch not necessary' );
 			return;
 		}
 
 		// defer fetch requests to avoid dispatcher conflicts
-		setTimeout( function() {
-			WordadsActions.fetchEarnings( site )
+		setTimeout( () => {
+			WordadsActions.fetchEarnings( site );
 		}, 0 );
-	},
+	}
 
-	toggleEarningsNotice: function( event ) {
+	updateSettings = () => {
+		this.setState( this.getSettingsFromStore() );
+	};
+
+	handleEarningsNoticeToggle = ( event ) => {
 		event.preventDefault();
-		this.setState( { showEarningsNotice: ! this.state.showEarningsNotice } );
-	},
+		this.setState( {
+			showEarningsNotice: ! this.state.showEarningsNotice
+		} );
+	};
 
-	toggleInfo: function( type, event ) {
+	handleInfoToggle = ( type ) => ( event ) => {
 		event.preventDefault();
 		switch ( type ) {
 			case 'wordads':
-				this.setState( { showWordadsInfo: ! this.state.showWordadsInfo } );
+				this.setState( {
+					showWordadsInfo: ! this.state.showWordadsInfo,
+				} );
 				break;
 			case 'sponsored':
-				this.setState( { showSponsoredInfo: ! this.state.showSponsoredInfo } );
+				this.setState( {
+					showSponsoredInfo: ! this.state.showSponsoredInfo,
+				} );
 				break;
 			case 'adjustment':
-				this.setState( { showAdjustmentInfo: ! this.state.showAdjustmentInfo } );
+				this.setState( {
+					showAdjustmentInfo: ! this.state.showAdjustmentInfo,
+				} );
```

Comment:
Did you consider folding these into one `case`? Your call as to whether it's improves readability or may reduce likelyhood of bugs.

```js
case 'adjustment':
case 'sponsored':
case 'wordads':
	this.setState( s => ( {
		[type]: ! s[type],
	} ) );
	break;
```